public/scripts/world-info.js

```diff
@@ -1898,6 +1898,44 @@ function getWIElement(name) {
     return wiElement;
 }
 
+/**
+ * Adds missing fields to WI entries that are present in the entry template, but not in the data.
+ * Additionally verify that array/object fields are of the expected type.
+ * @param {any[]} data WI entries
+ * @returns {any[]} Data with backfilled fields
+ */
+function addMissingWorldInfoFields(data) {
+    data.forEach((entry) => {
+        // Add missing fields from the template
+        Object.entries(newWorldInfoEntryTemplate).forEach(([key, value]) => {
+            if (!Object.hasOwn(entry, key)) {
+                entry[key] = structuredClone(value);
+            }
+        });
+
+        // Ensure that the key is always an array
+        if (!Array.isArray(entry.key)) {
+            entry.key = [];
```

Comment:
You *could* do a console log here of the original `entry.key` value, otherwise its a silent overwrite of data.
But considering its invalid anyway, I think its also okay not to.
(Some for the checks below)